File: src/main/java/com/cloudpi/cloudpi_backend/user/entities/UserPermissionEntity.java
package com.cloudpi.cloudpi_backend.user.entities;

import com.cloudpi.cloudpi_backend.security.AccountType;
import lombok.*;
import org.hibernate.Hibernate;
import javax.persistence.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
public class UserPermissionEntity {
    @Id
    @GeneratedValue
    private Long id;
    @Column
    private String permission;
    @ManyToOne
    @JoinColumn(name = "user_id")
    private UserEntity user;



    public void setPermission(String permission) {
            for (var accountType : AccountType.values()) {
                if (accountType.toString().equals(permission)) {
                    throw new RuntimeException();
                }
            }

            this.permission = permission;
    }
}

File: src/main/java/com/cloudpi/cloudpi_backend/user/entities/UserPermissionEntity.kt
package com.cloudpi.cloudpi_backend.user.entities

import com.cloudpi.cloudpi_backend.security.AccountType
import org.hibernate.Hibernate
import javax.persistence.*

@Entity
public class UserPermissionEntity {
    private permission: String,
    @ManyToOne
    @JoinColumn(name = "user_id")
    val user: UserEntity

    @Id
    @GeneratedValue
    var id: Long = 0


    var permission: String = permission
        set(value) {
            for (accountType in AccountType.values()) {
                if (accountType.toString() == value) {
                    throw Exception()
                }
            }

            field = value
        }
}